### PR TITLE
Add support for Ollama batch endpoint

### PR DESCRIPTION
**Pull request by antas-marcin**
### What's being changed:

This PR:
- adds support for Ollama's `/api/embed` endpoint (`/api/embeddings` endpoint is deprecated)
- adds support for batch in `tex2vec-ollama` module
- adds batch e2e Ollama vectorizer module tests

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [ ] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

File: modules/text2vec-ollama/clients/ollama.go [hunk 1]
```diff
@@ -85,11 +105,11 @@ func (v *ollama) vectorize(ctx context.Context, input string,
...
 
 func (v *ollama) parseEmbeddingsResponse(statusCode int,
-	bodyBytes []byte, input string,
-) (*ent.VectorizationResult, error) {
+	bodyBytes []byte, input []string,
+) (*modulecomponents.VectorizationResult, error) {
 	var resBody embeddingsResponse
 	if err := json.Unmarshal(bodyBytes, &resBody); err != nil {
-		return nil, errors.Wrap(err, fmt.Sprintf("unmarshal response body. Got: %v", string(bodyBytes)))
+		return nil, fmt.Errorf("unmarshal response body. Got: %v: %w", string(bodyBytes), err)
```
Comment (dirkkul) [inline modules/text2vec-ollama/clients/ollama.go:112]: please use wrapf instead. You loose the error-type with Errorf. Probably doesn't matter here, but that is a good idea in general

File: modules/text2vec-ollama/clients/ollama.go [hunk 2]
```diff
@@ -45,24 +49,40 @@ func New(timeout time.Duration, logger logrus.FieldLogger) *ollama {
...
 
-func (v *ollama) vectorize(ctx context.Context, input string,
-	config ent.VectorizationConfig,
-) (*ent.VectorizationResult, error) {
+func (v *ollama) GetApiKeyHash(ctx context.Context, config moduletools.ClassConfig) [32]byte {
+	return sha256.Sum256([]byte("ollama"))
+}
+
+func (v *ollama) GetVectorizerRateLimit(ctx context.Context, cfg moduletools.ClassConfig) *modulecomponents.RateLimits {
+	return &modulecomponents.RateLimits{}
```
Comment (dirkkul) [inline modules/text2vec-ollama/clients/ollama.go:70]: I would put some high dummy values in

File: modules/text2vec-ollama/module.go [hunk 3]
```diff
@@ -26,18 +26,30 @@ import (
...
+	// the encoding is different than OpenAI, but the code is not available in Go and too complicated to port.
+	// using 30% more than the OpenAI model is a rough estimate but seems to work
+	TokenMultiplier:    1.3,
+	MaxTimePerBatch:    float64(10),
+	MaxObjectsPerBatch: 512, // Info from jina
+	// real limit is 8192, but the vectorization times go up by A LOT if the batches are larger
+	MaxTokensPerBatch: func(cfg moduletools.ClassConfig) int { return 2500 },
+	HasTokenLimit:     true,
+	ReturnsRateLimit:  false,
+}
```
Comment (dirkkul) [inline modules/text2vec-ollama/module.go:45]: these seem to be copied from the jina module.

I think:
- TokenMultiplier: 0
- MaxObjectsPerBatch ????
- MaxTokensPerBatch High dummy value. If the multiplier is 0 any number above 0 should be fine though

File: modules/text2vec-ollama/clients/ollama.go (hunk 2 above)
Comment (antas-marcin) [reply modules/text2vec-ollama/clients/ollama.go]: pushed a proposal

File: modules/text2vec-ollama/module.go (hunk 3 above)
Comment (antas-marcin) [reply modules/text2vec-ollama/module.go:42]: pushed a proposal too but would like to have a review if proposed values are ok

Comment (dirkkul) [reply modules/text2vec-ollama/module.go:42]: I think we should do HasTokenLimit:  false,

**Review by dirkkul — APPROVED** [review]

**Review by dirkkul — APPROVED** [review]

**Pull request merged**